src/cpp/CMakeLists.txt

```diff
@@ -19,6 +19,43 @@ target_include_directories(kittie
 	$<INSTALL_INTERFACE:${CMAKE_INSTALL_PREFIX}/include>
 )
 
-install(TARGETS kittie DESTINATION ${CMAKE_INSTALL_PREFIX}/lib)
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/kittie.h DESTINATION ${CMAKE_INSTALL_PREFIX}/include)
+include(GNUInstallDirs)
+set(INSTALL_CONFIGDIR ${CMAKE_INSTALL_LIBDIR}/cmake/kittie)
 
+install(TARGETS kittie 
+	EXPORT kittie-targets 
+	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/kittie.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+
+install(EXPORT kittie-targets
+    FILE
+        kittieTargets.cmake
+    NAMESPACE
+        kittie::
+    DESTINATION
+        ${INSTALL_CONFIGDIR}
+)
+
+include (CMakePackageConfigHelpers)
+write_basic_package_version_file(
+    ${CMAKE_CURRENT_BINARY_DIR}/kittieConfigVersion.cmake
+    VERSION ${PROJECT_VERSION}
+    COMPATIBILITY AnyNewerVersion
+)
+
+configure_package_config_file(${CMAKE_CURRENT_LIST_DIR}/kittieConfig.cmake.in
+    ${CMAKE_CURRENT_BINARY_DIR}/kittieConfig.cmake
+    INSTALL_DESTINATION ${INSTALL_CONFIGDIR}
+)
+
+install(FILES
+    ${CMAKE_CURRENT_BINARY_DIR}/kittieConfig.cmake
+    ${CMAKE_CURRENT_BINARY_DIR}/kittieConfigVersion.cmake
+    DESTINATION ${INSTALL_CONFIGDIR}
+)
+
+export(EXPORT kittie-targets
```

Comment:
How is this different than the other export? They look like they're using the same target, but I don't really know how the `EXPORT` command works.